### PR TITLE
BOOKKEEPER-919: Auditor is sometimes marking as failed a bookie switc…

### PR DESCRIPTION
**Pull request by merlimat**
…hing from available to read-only mode

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcher.java [hunk 1]
```diff
@@ -363,14 +363,18 @@ public void processResult(int rc, String path, Object ctx, List<String> children
...
                         // listen children changed event from ZooKeeper
                         if (event.getType() == EventType.NodeChildrenChanged) {
                             listener.availableBookiesChanged();
                         }
                     }
                 });
+
+                // Update the list of read-only bookies
+                HashSet<BookieSocketAddress> newReadOnlyBookies = convertToBookieAddresses(children);
+                readOnlyBookies = newReadOnlyBookies;
```
Comment (jiazhai) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcher.java]: The change looks good. :).
Just for curious here, For this two lines, Is it OK to make the assignment directly as  "readOnlyBookies = convertToBookieAddresses(children)"?

Comment (merlimat) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcher.java]: Yes, it's exactly the same. I just left it in the same form it was already written at line 404 in the same file

**Comment by eolivelli** [comment]
+1
On my machine the patch fixes the AuditorLedgerCheckerTest test

**Comment by sijie** [comment]
LGTM. +1

**Pull request closed without merging**